### PR TITLE
Add sidecar injector to pilot

### PR DESCRIPTION
**Pull request by howardjohn**
Part of
https://docs.google.com/document/d/1jO2oM9zdm9h7tUqOpnAG3dh1xavtW1L7zQ5LyraYRAs.

Right now this change is not activated, because the default Pilot
deployment will not have the injector mounted. This will require changes
to istio/installer to set this up. It is intentional that istio/istio
helm installer is not changed here, as we probably want to leave that
untouched.

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -1424,6 +1442,41 @@ func (s *Server) initEventHandlers() error {
...
+		if err != nil {
+			return err
+		}
+		parameters := inject.WebhookParameters{
+			ConfigFile: filepath.Join(injectPath, "config"),
+			ValuesFile: filepath.Join(injectPath, "values"),
+			MeshFile:   args.Mesh.ConfigFile,
+			CertFile:   filepath.Join(dir, "cert-chain.pem"),
+			KeyFile:    filepath.Join(dir, "key.pem"),
+			Port:       args.InjectionOptions.Port,
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: How do we enable healthcheck?

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -1424,6 +1442,41 @@ func (s *Server) initEventHandlers() error {
...
+		}
+
+		wh, err := inject.NewWebhook(parameters)
+		if err != nil {
+			return fmt.Errorf("failed to create injection webhook: %v", err)
+		}
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go wh.Run(stop)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: Could not find the cert reconciling loop

Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: This is intended. See https://github.com/istio/istio/issues/17061 for more info. The webhook cert will be manged by istioctl/the operator. The cert has a 5 year TTL anyways so we don't really need such active management in pilot?

File: pilot/pkg/bootstrap/server.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: Why would we need it? We already have pilot health check, the webhook check doesn't do anything especially useful?

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: They are different modules, the healthcheck should cover all, and aggregate all healthchecks

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: Is the ttl configurable?

Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: I don't think so @lei-tang can you verify

File: pilot/pkg/bootstrap/server.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: Ok good point, will make aggregate check

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (lei-tang) [reply pilot/pkg/bootstrap/server.go]: The DNS certificate signed by Kubernetes CA has a 5 year TTL and is not configurable.

**Pull request merged**